iceoryx_utils/include/iceoryx_utils/posix_wrapper/timer.hpp

```diff
@@ -25,6 +25,7 @@
 #include <condition_variable>
 #include <cstdint>
 #include <ctime>
+#include <functional>
```

Comment:
Why did you add the functional header?

Reply:
Used to be implicitly included via `optional` before.